### PR TITLE
Cookies now always redirects to Login/AccessDenied Paths

### PR DESCRIPTION
**Pull request by HaoK**
We default values for the paths if they are not set in options

cc @Tratcher

### CODE REVIEW

File: src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationMiddleware.cs [hunk 1]
```diff
@@ -40,6 +40,18 @@ public class CookieAuthenticationMiddleware : AuthenticationMiddleware<CookieAut
             {
                 Options.CookieManager = new ChunkingCookieManager(urlEncoder);
             }
+            if (!Options.LoginPath.HasValue)
+            {
+                Options.LoginPath = CookieAuthenticationDefaults.LoginPath;
+            }
+            if (!Options.LogoutPath.HasValue)
+            {
+                Options.LogoutPath = CookieAuthenticationDefaults.LogoutPath;
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationMiddleware.cs]: Should a logout redirect still be optional? SignOut("Cookies") primary job is to delete the cookie.

Comment (HaoK) [inline src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationMiddleware.cs]: I don't think this one is a redirect. This one just specified what uri will enable the return uri to trigger
A redirect so I think it's ok to have a default.

They still have to send a query string with return url on it to the actual logout path and call sign out there for this to have any effect

**Comment by PinpointTownes** [comment]
Curious: won't this change conflict with social middleware configured to use the automatic mode?

Having both an OIDC middleware and a cookie middleware using active authentication is very common: in this case, you DON'T want your users to be redirected to a local path (they should be directly redirected to the OIDC provider).

**Comment by HaoK** [comment]
Cookies is supposed to be first. Which means last in the chain. If you put one of the earlier ones I in automatic and it handles the request it should stop cookies from getting hit.

File: test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs [hunk 2]
```diff
@@ -721,6 +715,22 @@ private Task SignOutAsWrong(HttpContext context)
...
 
         [Fact]
+        public async Task ChallengeDoesNotSet401OnUnauthorized()
+        {
+            var server = TestServer.Create(app =>
+            {
+                app.UseCookieAuthentication();
+                app.Run(async context => {
+                    await Assert.ThrowsAsync<InvalidOperationException>(() => context.Authentication.ChallengeAsync());
+                    context.Response.StatusCode = 200;
```
Comment (Tratcher) [inline test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs]: You should be able to remove this line, it may create false positives.

**Comment by Tratcher** [comment]
:shipit:

**Comment by HaoK** [comment]
b883920befce4afaac89ec031d30d868a48df666

**Pull request closed without merging**